R/keyword_quote_linter.R

```diff
@@ -52,6 +52,11 @@
 #   //expr[expr[SYMBOL_FUNCTION_CALL]]/SYMBOL_SUB[starts-with(text(), '`')]
 #   //expr[expr[SYMBOL_FUNCTION_CALL]]/STR_CONST[{is_quoted(text())}]
 keyword_quote_linter <- function() {
+  # Check if a string could be assigned as an R variable.
+  #
+  # See [make.names()] for the description of syntactically valid names in R.
+  is_valid_r_name <- function(x) make.names(x) == x
```

Comment:
This is always used with `x = get_r_string(...)`. Worth pulling it inside the function?